### ISSUE TITLE
💡 RFC:  comments inside a tag (component or html)

### ISSUE THREAD

**Issue by hans-d** [snowpackjs/astro#1833]
### Background & Motivation

When writing astro files, it can be helpful to add documentation to make intentions more clear. In the frontmatter this can be done(`//` or `\* *\`), as well as using regular HTML comment tags (`<!-- -->`). 
Inside tags, esp with certain attributes, this might be usefull as well.

The comments are only needed in the astro file, not needed on rendering (as html does not have such an option).

 These comments can be ignored on rendering, so they are only in .astro files

https://discord.com/channels/830184174198718474/872579324446928896/909389002820431872

### Possible solutions

Allow adding `/* */` style comments inside tags. This would mimic other comment systems that enables comments within another expression.

```
<div
   attr=value /* some remark */
   empty /* to be used for xyz */
   other=test
   /* TODO: something */
>
```

### Risks, downsides, and/or tradeoffs

- Using only '//' would cause issues if reformatting is done (eg prettifier)
- Dealing with various exceptions, like:
  - '/*' inside quotes, eg '<div a="/* no comment! */">
  - `<div /* this is something> <div that might happen */ > <div /* as /* is this */ >`, (imho this should be dealt with by the user, just like `<!--`  and other javascript comment issues

### Detailed Design

tbd

### Help make it happen!

- [ ] I am willing to submit a PR to implement this change.
- [X] I am willing to submit a PR to implement this change, but would need some guidance.
- [ ] I am not willing to submit a PR to implement this change.

**Comment by natemoo-re** [comment]
I know it's not exactly what you're asking for here, but IMO this is superseded by the bugfix in https://github.com/snowpackjs/astro-compiler-next/pull/149

We can't support HTML comments inside of a tag because the HTML spec disallows it. We can't support plain `/* comments */` in Astro files because JS-style comments are not supported in HTML.

What we _can_ do (and was a bug that we didn't) is support JS-style comments inside of JS expressions, which unblocks your use case.

Your example could become:

```astro
<div
   attr=value {/* some remark */}
   empty {/* to be used for xyz */}
   other=test
   {/* TODO: something */}
>
```

**Comment by hans-d** [comment]
This would be perfect as well, and is currently not working
